File: runtime/flutter/painting/hSLColor.ts

import { IColor } from "../../dart/ui/color";
declare const flutter: {
painting: {
hSLColorFromColor: (color: IColor) => IHSLColor;
hSLColorLerp: (a: IHSLColor | undefined, b: IHSLColor | undefined, t: number) => IHSLColor | undefined
}
};
export interface IHSLColor


{
alpha: number;
hue: number;
saturation: number;
lightness: number;
withAlpha: (alpha: number) => IHSLColor;
withHue: (hue: number) => IHSLColor;
withSaturation: (saturation: number) => IHSLColor;
withLightness: (lightness: number) => IHSLColor;
toColor: () => IColor;
getHashCode: () => number;
toString: () => string;
}export class HSLColor


{    public readonly alpha: number = undefined as any;
    public readonly hue: number = undefined as any;
    public readonly saturation: number = undefined as any;
    public readonly lightness: number = undefined as any;
public static fromColor(color: IColor) : IHSLColor{
return flutter.painting.hSLColorFromColor(color);
}public static lerp(a: IHSLColor | undefined, b: IHSLColor | undefined, t: number) : IHSLColor | undefined{
return flutter.painting.hSLColorLerp(a, b, t);
}    private readonly _dart_withAlpha: (alpha: number) => IHSLColor = undefined as any;
    private readonly _dart_withHue: (hue: number) => IHSLColor = undefined as any;
    private readonly _dart_withSaturation: (saturation: number) => IHSLColor = undefined as any;
    private readonly _dart_withLightness: (lightness: number) => IHSLColor = undefined as any;
    private readonly _dart_toColor: () => IColor = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
public withAlpha(alpha: number) : IHSLColor {
    return this._dart_withAlpha(alpha);
}
public withHue(hue: number) : IHSLColor {
    return this._dart_withHue(hue);
}
public withSaturation(saturation: number) : IHSLColor {
    return this._dart_withSaturation(saturation);
}
public withLightness(lightness: number) : IHSLColor {
    return this._dart_withLightness(lightness);
}
public toColor() : IColor {
    return this._dart_toColor();
}
public getHashCode() : number {
    return this._dart_getHashCode();
}
public toString() : string {
    return this._dart_toString();
}
}

File: lib/cfr/builtins/libs/flutter/painting/hSLColor.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/colors.dart' as _898f;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedHSLColor extends _36c2.VMManagedBox<_898f.HSLColor> {VMManagedHSLColor({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'alpha'
] = vmObject.alpha;
table  [
'hue'
] = vmObject.hue;
table  [
'saturation'
] = vmObject.saturation;
table  [
'lightness'
] = vmObject.lightness;
table  [
'withAlpha'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSLColor>(object: vmObject.withAlpha(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withHue'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSLColor>(object: vmObject.withHue(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withSaturation'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSLColor>(object: vmObject.withSaturation(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withLightness'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSLColor>(object: vmObject.withLightness(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toColor'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Color>(object: vmObject.toColor(), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(),]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _898f.HSLColor vmObject;

 }

void loadHSLColor({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'hSLColorFromColor'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSLColor>(object: _898f.HSLColor.fromColor(_36c2.maybeUnBoxAndBuildArgument<_a643.Color, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'hSLColorLerp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

final returnValue = _898f.HSLColor.lerp(_36c2.maybeUnBoxAndBuildArgument<_898f.HSLColor?, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_898f.HSLColor?, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState), luaCallerArguments  [
3
]?.toDouble());if(returnValue!= null){return [_36c2.maybeBoxObject<_898f.HSLColor?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
_36c2.registerBoxer<_898f.HSLColor>(boxer: ({required _898f.HSLColor vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedHSLColor(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
